src/main/java/com/uid2/optout/partner/OptOutPartnerEndpoint.java

```diff
@@ -33,36 +42,53 @@ public String name() {
     @Override
     public Future<Void> send(OptOutEntry entry) {
         return this.retryingClient.send(
-            req -> {
-                for (String queryParam : this.config.queryParams()) {
+            (URI uri, HttpMethod method) -> {
```

Comment:
I know you didn't write this from scratch so this is optional. 
Perhaps we should pre-calculate those extra headers/query strings/... on creation instead of - re-parsing them on every request? With this setup we will also not know if anything is wrong until the call actually gets triggered so a bug or bad config could sit there for days or weeks after change.

Reply:
For building the query params it's also better to use a https://docs.oracle.com/cd/E39271_01/studio.300/studio_api_ref/com/endeca/util/UrlBuilder.html or similar - it will also handle things like encoding of params which is not done now.

Reply:
Use URIBuilder, done